src/test/cc/wfanet/panelmatch/common/crypto/aes_with_hkdf_test.cc

```diff
@@ -63,11 +97,69 @@ TEST(AesWithHkdfTest, emptyKeyDecrypt) {
               StatusIs(absl::StatusCode::kInvalidArgument, ""));
 }
 
-// TODO(enfreck): Implement the following cases
-//  - Encrypt/Decrypt
-//  - Null input
-//  - Different key with same string are different things
-//  - Same key with different strings are different things
+// Tests that different keys with the same string return different values
+TEST(AesTest, differentKeySameStringEncrypt) {
+  std::unique_ptr<Hkdf> hkdf = GetSha256Hkdf();
+  std::unique_ptr<Aes> aes = GetAesSivCmac512();
+  SecretData key_1 = SecretDataFromStringView("key1");
+  SecretData key_2 = SecretDataFromStringView("key2");
+  std::string_view plaintext = "Some data to encrypt.";
+  AesWithHkdf aes_hkdf(std::move(hkdf), std::move(aes));
+  ASSERT_OK_AND_ASSIGN(std::string cyphertext_1,
+                       aes_hkdf.Encrypt(plaintext, key_1));
+  ASSERT_OK_AND_ASSIGN(std::string cyphertext_2,
+                       aes_hkdf.Encrypt(plaintext, key_2));
+  EXPECT_NE(cyphertext_1, cyphertext_2);
+}
+
+// Tests that decrypting with a different key than encryption gives an error
+TEST(AesTest, differentKeyDecrypt) {
+  std::unique_ptr<Hkdf> hkdf = GetSha256Hkdf();
+  std::unique_ptr<Aes> aes = GetAesSivCmac512();
+  SecretData key_1 = SecretDataFromStringView("key1");
+  SecretData key_2 = SecretDataFromStringView("key2");
+  std::string_view plaintext = "Some data to encrypt.";
+  AesWithHkdf aes_hkdf(std::move(hkdf), std::move(aes));
+  ASSERT_OK_AND_ASSIGN(std::string cyphertext,
+                       aes_hkdf.Encrypt(plaintext, key_1));
+  auto decrypted = aes_hkdf.Decrypt(cyphertext, key_2);
+  EXPECT_THAT(decrypted.status(),
+              StatusIs(absl::StatusCode::kInvalidArgument, ""));
+}
+
+// Tests that the same key with different strings return different values
+TEST(AesTest, sameKeyDifferentStringEncrypt) {
+  std::unique_ptr<Hkdf> hkdf = GetSha256Hkdf();
+  std::unique_ptr<Aes> aes = GetAesSivCmac512();
+  SecretData key = SecretDataFromStringView("key");
+  std::string_view plaintext_1 = "Some data to encrypt.";
+  std::string_view plaintext_2 = "Additional data to encrypt.";
+  AesWithHkdf aes_hkdf(std::move(hkdf), std::move(aes));
+  ASSERT_OK_AND_ASSIGN(std::string cyphertext_1,
+                       aes_hkdf.Encrypt(plaintext_1, key));
+  ASSERT_OK_AND_ASSIGN(std::string cyphertext_2,
+                       aes_hkdf.Encrypt(plaintext_2, key));
+  EXPECT_NE(cyphertext_1, cyphertext_2);
+}
+
+// Tests that the same key with different strings return different values
```

Comment:
Same as above -- comment describing the test is the same as for the previous test.